### PR TITLE
cpu/lm4f120: ADC support

### PR DESCRIPTION
**Pull request by dkm**
Only ADC_0 and no precision support.

### CODE REVIEW

**Comment by emmanuelsearch** [comment]
Seems to work. Flashed the perish_adc test and this is the output(which reacts with bigger values when the board is in my hands, thus bigger temperatures).

```
Welcome to pyterm!
Type '/exit' to exit.
2016-03-07 16:48:19,522 - INFO # Values: ADC_0-CH0:  262  ADC_0-CH1:  257  ADC_0-CH2:  257  ADC_0-CH3:  257  ADC_0-CH4:  257  ADC_0-CH5:  256  ADC_0-CH6:  257  ADC_0-CH7:  257  ADC_0-CH8:  258  ADC_0-CH9:  257  ADC_0-CH10:  256  ADC_0-CH11:  256  
2016-03-07 16:48:19,527 - INFO # Values: ADC_0-CH0:  265  ADC_0-CH1:  258  ADC_0-CH2:  256  ADC_0-CH3:  256  ADC_0-CH4:  257  ADC_0-CH5:  258  ADC_0-CH6:  257  ADC_0-CH7:  257  ADC_0-CH8:  257  ADC_0-CH9:  257  ADC_0-CH10:  259  ADC_0-CH11:  260  
2016-03-07 16:48:19,530 - INFO # Values: ADC_0-CH0:  265  ADC_0-CH1:  257  ADC_0-CH2:  256  ADC_0-CH3:  256  ADC_0-CH4:  259  ADC_0-CH5:  257  ADC_0-CH6:  257  ADC_0-CH7:  257  ADC_0-CH8:  258  ADC_0-CH9:  257  ADC_0-CH10:  259  ADC_0-CH11:  257  
```

**Comment by emmanuelsearch** [comment]
A detail: when building there is a warning thrown: 

```
There are unsatisfied feature requirements: periph_adc

EXPECT ERRORS!
```

**Comment by dkm** [comment]
Ah, sorry for that. I saw this error but it was still compiling/working during the tests and I forgot to go back on this.
I have the feature declared in Makefile.features. What am I missing ?

**Comment by dkm** [comment]
Fixed the stupid typo in Makefile.features. Compiles correctly now.

File: boards/ek-lm4f120xl/Makefile.features [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_adc
```
Comment (emmanuelsearch) [inline boards/ek-lm4f120xl/Makefile.features]: Easy. It's just a typo. It's PROVIDED instead of PREVIDED ;)

**Comment by dkm** [comment]
Yes, I've already pushed the fix :)

**Comment by emmanuelsearch** [comment]
Yup, seems we cross-fired on this one ;)

**Comment by emmanuelsearch** [comment]
Uncrustify does not seem to complain. ACK.

**Comment by dkm** [comment]
ok, great :)

**Comment by emmanuelsearch** [comment]
weird CI fails. Will kick again to serif it sorts itself out.

**Comment by dkm** [comment]
not sure for RIOT-CI as I can't find a timestamp in the logs, but travis has not been executed again :(

**Comment by DipSwitch** [comment]
It has now. Go.

**Comment by dkm** [comment]
Thanks !

**Comment by haukepetersen** [comment]
Guys, next time please check the style of the PR manually before merging, there are quite some issues in this PR (alignment, c++ style comments...)

**Comment by haukepetersen** [comment]
also the `precision` setting is never used, so which precision is this driver using?

**Comment by dkm** [comment]
Sorry to head that. For alignment, I thought that running uncrustify would take care of that.
I will remove the c++ comments (what is the rationale behind this choice ? I see that RIOT is using c99 so any conforming compiler should be able to handle them).
I will make clear in the code what precision is used.

I will push a PR quickly to fix this.

**Comment by DipSwitch** [comment]
This is already fixed in #4430 if I'm correct?

**Comment by dkm** [comment]
You're right :)

**Comment by haukepetersen** [comment]
yes, I took care of it :-)

**Pull request merged**